File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = Integer.MAX_VALUE;
        int curr = 0;
        int max = 0;

        for(int i = 0; i< prices.length; i++)
        {
            System.out.println("yes");
            buy = Math.min(prices[i], buy);
            curr = prices[i]-buy;
            max = Math.max(max, curr);
        }
        return max;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        maxprofit = 0
        currprofit = 0
        buy = 0
        n = len(prices)
        for i in range(1, n):
            if prices[i] < prices[buy]:
                buy = i
                maxprofit = max(maxprofit, currprofit)
                currprofit = 0
            elif prices[i] > prices[buy]:
                currprofit = prices[i] - prices[buy]
                maxprofit = max(maxprofit, currprofit)
        return max(maxprofit, currprofit)
